content/context-resource/index.md

```diff
@@ -0,0 +1,320 @@
+---
+rfc: 0018
+start_date: 2018-08-13
+pr: openregister/registers-rfcs#31
+status: draft
+---
+
+# Context resource
+
+## Summary
+
+This RFC proposes a context resource to surface the metadata of the register.
+
+Meanwhile there is no versioning defined for metadata treat this as single
+versioned.
+
+Dependencies:
+
+* [RFC0013: Multihash](https://github.com/openregister/registers-rfcs/pull/26)
+
+
+## Motivation
+
+Currently, the register metadata is not exposed consistently. Some is exposed
+via the register resource, some via the field register and some from the
+register register.
+
+Also, the current specification and implementation don't agree on what is the
+register resource.
+
+The main driver for this RFC is to have a computed resource for the latest
+metadata in a way that makes the register self contained (i.e. a user should
+not need to interact with any other register to get the metadata for a
+register).
+
+
+## Explanation
+
+The **context** is the metadata snapshot that apply to a given log size.
+
+### Definition
+
+```elm
+type Context =
+  { id : Name
+  , copyright : String
+  , custodian : Maybe String
+  , description : Maybe String
+  , hashingAlgorithm : HashingAlgorithm
+  , licence : Maybe String
+  , rootHash : Hash
+  , schema : Schema
+  , stats : Stats
+  , status : Status
+  , title : Maybe String
+  }
+```
+
+See each section below of the explanation for each attribute.
+
+
+#### Id
+
+* Type: Name (This is defined in the specification as `[a-z][a-z0-9-]*`, the
+  future specification will make it more clear).
+
+The register identifier. The current implications for this value make it act
+as a unique identifier for the environment is belongs to. This RFC doesn't
+pretend to change this fact.
+
+
+#### Copyright
+
+* Type: String
+
+The register copyright. E.g. `© Crown copyright`. Similar to the `copyright`
+field in the Register register but required to be informed.
+
+
+#### Custodian
+
+* Type: String
+
+The same value currently exposed in the Register resource.
+
+
+#### Description
+
+* Type: Optional String
```

Comment:
Any reason why this can't be mandatory if we return it here? A register should always have a human readable description, otherwise how do people know to use it?

Reply:
Well, ideally yes and it is something we will probably want to enforce for registers part of the our catalogue but I can foresee registers not having a description at early/immature stages.